google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/DefaultCloudSdkService.java

```diff
@@ -40,4 +83,206 @@ public String getCloudSdkHomePath() {
   public void setCloudSdkHomePath(String cloudSdkHomePath) {
     propertiesComponent.setValue(CLOUD_SDK_PROPERTY_KEY, cloudSdkHomePath);
   }
+
+  @Nullable
+  private Path getJavaToolsBasePath() {
```

Comment:
let's leave the nullable returns as is then. I don't think logging.error makes sense since as you say, these methods can be called when the user doesn't have an SDK set.